From 214f39f542d01dd6ea0250789f1ca38c6464f4a9 Mon Sep 17 00:00:00 2001
From: Subhankar Paul <subhankar.paul@intel.com>
Date: Tue, 17 May 2016 10:28:32 +0530
Subject: [PATCH 1/3] Audio: Allocate compress offload ddr ring buffer in DMA
 zone

Since ddr ring buffer memory is used for DMA transfer, kmalloc
need to allocate memory in DMA zone.

Change-Id: Ia0f95a90d1e00c562dc7a55745101ecb6ea3f34e
Signed-off-by: Subhankar Paul <subhankar.paul@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-31151
Reviewed-on: https://android.intel.com:443/506409
---
 sound/core/compress_offload.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/core/compress_offload.c b/sound/core/compress_offload.c
index a99a389..4821122 100644
--- a/sound/core/compress_offload.c
+++ b/sound/core/compress_offload.c
@@ -486,7 +486,7 @@ static int snd_compr_allocate_buffer(struct snd_compr_stream *stream,
 		 * the data from core
 		 */
 	} else {
-		buffer = kmalloc(buffer_size, GFP_KERNEL);
+		buffer = kmalloc(buffer_size, GFP_KERNEL | GFP_DMA);
 		if (!buffer)
 			return -ENOMEM;
 	}
-- 
1.9.1

